src/tuple.cr

```diff
@@ -175,6 +175,7 @@ struct Tuple
   # tuple.at(3) { 10 } # => 10
   # ```
   def at(index : Int)
+    index += size if index < 0
```

Comment:
The reason that I don't use `Indexable#check_index_out_of_bound` is type inference works correctly.

If I rewrite it with using `check_index_out_of_bouds`, this method will become like below:

```crystal
  def at(index : Int)
    index = check_index_out_of_bounds { return yield }
    {% for i in 0...T.size %}
      return self[{{i}}] if {{i}} == index
    {% end %}
    raise "BUG: unreachable"
  end
```

The last `raise` is important. Without this, `at` method result type is inferred as nilable. And I think `return yield` inside block looks bad, so I don't use `check_index_out_of_bounds`.